src/reducers/marksReducer.js

```diff
@@ -1,42 +1,31 @@
-import keys from '../data/key';
+import keys from "../data/key";
 
-const initToggleState = {
-  status: false
-};
-
-const initModalState = {
-  showModal: false
-};
-
-const initMarkerState = {
+const INITIAL_STATE = {
+  status: false,
+  showModal: false,
   marks: []
 };
 
-export const marksFetchReducer = (state = initMarkerState.marks, action) => {
-  switch (action.type) {
-    case keys.MARKS_FETCH_DATA_SUCCESS:
-      return action.marks;
-    case keys.MARK_ADD_DATA_SUCCESS:
-      return state.concat(action.marks);
-    default:
-      return state;
-  }
-};
-
-export const marksToggleReducer = (state = initToggleState.status, action) => {
-  switch (action.type) {
-    case keys.TOGGLE_MARKER:
-      return action.status;
-    default:
-      return state;
-  }
-};
-
-export const addMarkerReducer = (state = initModalState.showModal, action) => {
-  switch (action.type) {
-    case keys.ADD_MARKER:
-      return action;
-    default:
-      return state;
-  }
-};
+export const marksReducer = (state = INITIAL_STATE, action) => {
+    switch (action.type) {
+      case keys.MARKS_FETCH_DATA_SUCCESS:
+        return {
+          ...state,
+          marks: action.marks
+        }
+      case keys.MARK_ADD_DATA_SUCCESS:
+        return state.concat(action.marks);
```

Comment:
Question: What does this expression differ from line 12 ~ 15?
```javascript
        return state.concat(action.marks);
```
```javascript
        return {
          ...state,
          marks: action.marks
        }
```

Reply:
First one is to initialize the map with the existing marks and the second one is to update the state when an new mark is added. the reason of using concat is because we don't wanna make another request everytime a mark is added. We just make an api request when you refresh the page which means when the components mount. This may affect async when more than one user add marks at the same time since it is not observing the changes but this has to be discussed for a better solution.

Reply:
@abdiG , @kc-leung Thank you for a quick reply. my question was about the way of expression instead of a feature. I meant we can express the way return 'state' both, 

```javascript
return { ...state, new_value }
```
or 
```javascript
return state.cancat(new_value)
```

isn't it?  I meant if both expressions are same, we'd better using the same way.

Reply:
I see what you mean. Yea you are right It is actually wrong and it should return always the complete state with the field or object key value you wanna modifiy